### PR TITLE
Add explanation for judge configuration

### PR DESCRIPTION
**Pull request by kiritofeng**

### CODE REVIEW

File: docs/judge/judge_configuration.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Judge Configuration
+The DMOJ judge is configured with a YAML file, which contains the runtimes, problems folders, and other information.
+
+## Runtimes
+
+The runtimes are configured with a `runtime` node. This can be automatically configured with `dmoj-autoconf`. As of writing, `dmoj-autoconf` is broken on Windows.
```
Comment (quantum5) [inline docs/judge/judge_configuration.md]: Windows is not even supported anymore.

File: docs/judge/judge_configuration.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+The judge's display name is configured with an `id` node. This is a string, and it should match the one on the site interface.
+
+## Key
+
+This key is used to validate your judge's connection to the bridge, and as such, should match the one on the site interface. This is configured with a `key` node.
+
+## Tempdir
+
+A Windows-only entry, configured with a `tempdir` node, which specifies a temperary directory for the judge to use. See [this](https://github.com/DMOJ/docs/blob/master/docs/judge/windows_installation.md#very-important-note) for more information.
```
Comment (quantum5) [inline docs/judge/judge_configuration.md]: Windows is no longer supported. How long ago did you write this?

Comment (kiritofeng) [reply docs/judge/judge_configuration.md]: Commit history says Dec 19, 2017.

File: docs/judge/judge_configuration.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Problems
+
+The problems are configured with `problem_storage_root`. There are three ways to configure this node:
+
+1. A single string: all grandchildren directories which include an `init.yml` will be treated as a problem directory.
+2. A list: all children of each element in the list will be treated as a problem directory.
+3. A dictionary: the key is an integer and the value is a string. The n-th level children of the string will be considered problem directories, where `n` is the entry's key.
+
+## Id
```
Comment (Xyene) [inline docs/judge/judge_configuration.md]: ID

File: docs/judge/judge_configuration.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Judge Configuration
+The DMOJ judge is configured with a YAML file, which contains the runtimes, problems folders, and other information.
+
+## Runtimes
+
+The runtimes are configured with a `runtime` node. This can be automatically configured with `dmoj-autoconf`.
```
Comment (Xyene) [inline docs/judge/judge_configuration.md]: Maybe mention that not all languages can be automatically configured.

File: docs/judge/judge_configuration.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Judge Configuration
+The DMOJ judge is configured with a YAML file, which contains the runtimes, problems folders, and other information.
```
Comment (Xyene) [inline docs/judge/judge_configuration.md]: Maybe provide a link to the sample, or quote it here.

**Pull request merged**